test/odbc/test_pg_datatype_bigint.cpp

```diff
@@ -0,0 +1,361 @@
+const string DATATYPE_NAME = "sys.bigint";
+
+vector<pair<string, string>> TABLE_COLUMNS = {
+    {COL1_NAME, DATATYPE_NAME + " PRIMARY KEY"},
+    {COL2_NAME, DATATYPE_NAME}
+  };
+
+class PSQL_DataTypes_Bigint : public testing::Test{
+
+  void SetUp() override {
+  }
+
+  void TearDown() override {
+
+    OdbcHandler test_setup;
+    test_setup.ConnectAndExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+};
+
+// helper function to convert string to equivalent C version of big int (long long int)
+long long int StringToBigInt(const string &value) {
+  return strtoll(value.c_str(), NULL, 10);
+}
+
+TEST_F(PSQL_DataTypes_Bigint, ColAttributes) {
+
+  const int LENGTH_EXPECTED = 20;
+  const int PRECISION_EXPECTED = 0;
+  const int SCALE_EXPECTED = 0;
+  const string NAME_EXPECTED = "int8";
+  
+  const int BUFFER_SIZE = 256;
+  char name[BUFFER_SIZE];
+  int length;
+  int precision;
+  int scale;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  // Create a table with columns defined with the specific datatype being tested. 
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS));
+  odbcHandler.CloseStmt();
+
+  // Select * From Table to ensure that it exists
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {"pk"}));
+
+  // Make sure column attributes are correct
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_LENGTH, // Get the length of the column (size of char in columns)
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &length);
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(length, LENGTH_EXPECTED);
+  
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_PRECISION, // Get the precision of the column
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &precision); 
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(precision, PRECISION_EXPECTED);
+
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_SCALE, // Get the scale of the column
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &scale); 
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(scale, SCALE_EXPECTED);
+
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_TYPE_NAME, // Get the type name of the column
+                          name,
+                          BUFFER_SIZE,
+                          NULL,
+                          NULL); 
+  ASSERT_EQ(string(name), NAME_EXPECTED);
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
```

Comment:
Don't we need to drop the created table here?
The TearDown() does it, but the next test is trying to create the table with the same name, in which case it would already exist.

Similar comment for other tests.

Reply:
TearDown() gets called after every test (not the whole test suite), so it will be dropped in between tests. I can create a drop statement for these tests though so we don't always depend on the TearDown() to cleanup.

Reply:
Ok, I wasn't sure. In this case we don't need the drop table at the end of each test.
We may still need drop table if exists before each create table though, in case the table has not been dropped because of an error.